### PR TITLE
Rework to device.js

### PR DESCRIPTION
**Pull request by brian-learningpool**
- Abstracted code into a getScreenWidth() function
- Corrected issue with width reported for iOS devices in landscape by adding getAppleScreenWidth() function

### CODE REVIEW

**Comment by brian-learningpool** [comment]
This resolves the reporting of the width in iOS devices as discussed at #1096.  In iOS the orientation is important in calculating the `screenWidth`, as when it's in portrait `screen.width` works fine, but when it's in landscape `screen.height` refers to the 'width'.  This solution should work both on a course embedded within an `iframe` and without.

@andrewhosgood, can you please test this resolves your issue?

**Comment by andrewhosgood** [comment]
+1 Tested but not within an iframe

**Comment by oliverfoster** [comment]
need to see if the iframe will scale properly as well, otherwise we end up in a weirder situation where the content width doesn't match the javascript measurement.

**Comment by oliverfoster** [comment]
+1 this needs to go in to fix the immediate issue, the iframe one still needs investigation

File: src/core/js/device.js [hunk 1]
```diff
@@ -69,6 +73,10 @@ define(function(require) {
         return /iPad|iPhone|iPod/.test(navigator.userAgent) && !window.MSStream;
     }
 
+    function getAppleScreenWidth() {
+        return (Math.abs(window.orientation) === 90) ? screen.height : screen.width;
```
Comment (oliverfoster) [inline src/core/js/device.js]: do we need 180 also?

Comment (andrewhosgood) [inline src/core/js/device.js]: You mean 270? It appeared to work in all orientations on the iPad I was testing on.

Comment (andrewhosgood) [inline src/core/js/device.js]: Actually ignore me, I missed the `Math.abs`!

Comment (oliverfoster) [inline src/core/js/device.js]: yea, that's exactly what i was meaning

Comment (oliverfoster) [inline src/core/js/device.js]: it goes -90 90 ?

Comment (oliverfoster) [inline src/core/js/device.js]: http://www.williammalone.com/articles/html5-javascript-ios-orientation/ yup, s'all good

File: src/core/js/device.js [hunk 2]
```diff
@@ -38,11 +38,15 @@ define(function(require) {
         return screenSize;
     }
 
+    function getScreenWidth() {
+        return isAppleDevice()
+            ? getAppleScreenWidth()
+            : window.innerWidth || $window.width()
```
Comment (tomgreenfield) [inline src/core/js/device.js]: The ending semi-colon has gone astray here. Feel free to ignore if it doesn't bother anyone!

Comment (brian-learningpool) [inline src/core/js/device.js]: Ah, nuts! Of course bothers me now...

**Comment by tomgreenfield** [comment]
+1

**Comment by oliverfoster** [comment]
+1

**Pull request merged**